kubectl-rook-ceph.sh

```diff
@@ -263,6 +265,83 @@ function run_purge_osd() {
     rook ceph osd remove --osd-ids=$1 --force-osd-removal=$force_removal"
 }
 
+####################################################################################################
+# 'kubectl rook-ceph debug commands
+####################################################################################################
+
+function update_deployment_spec_command(){
+  deployment_spec=$1
+  main_command="[\"sleep\",\"infinity\"]"
+  deployment_spec=$(jq .template.spec.containers[0].command="$main_command" <<< $deployment_spec)
+  deployment_spec=$(jq .template.spec.containers[0].args=[] <<< $deployment_spec)
+}  
+
+function update_deployment_spec_image(){
+  deployment_spec=$1
+  alternalte_image=$2
+  deployment_spec=$(jq .template.spec.containers[0].env[0].value="$alternalte_image" <<< $deployment_spec)
+  deployment_spec=$(jq .template.spec.containers[0].image="$alternalte_image" <<< $deployment_spec)
+}
+
+function run_create_debug(){
+  [[ -z "${1:-""}" ]] && fail_error "Missing 'mon/osd/mgr/mds' deployment name"
+  deployment_name=$1
+  echo "create debug mode for $deployment_name"
+  if output=$($TOP_LEVEL_COMMAND --namespace "$ROOK_OPERATOR_NAMESPACE" get deployment "$deployment_name"); then
+    echo "deployment $deployment_name exist, $output" 
+  else  
+    echo "deployment $deployment_name doesn't exist"
+    exit 1
+  fi
+  
+  # copy the deployment spec before scaling it down
+  deployment_spec=$($TOP_LEVEL_COMMAND get  deployments --namespace "$ROOK_OPERATOR_NAMESPACE"  "$deployment_name"   -o json | jq -r ".spec")
+  
+  # update the deployment_spec main container to be only a placeholder,
+  # also update image if alternate-image is passed
+  if [[ -z "${2:-""}" ]]; then
+    echo "alternate-image is not passed, so using the same container image and updating main container command"
+    update_deployment_spec_command "$deployment_spec"
+  else
+    echo "updating the alternalte image and main container command"
+    update_deployment_spec_command "$deployment_spec"
+    update_deployment_spec_image "$deployment_spec" "$2"
+  fi  
```

Comment:
What if this instead did something like this...

```suggestion
  if [[ -n "$desired_image" ]]; then 
    echo "setting debug image to $desired_image"
    update_deployment_spec_image "$deployment_spec" "$desired_image"
  fi
  echo "setting debug command to main container"
```

If "setting debug image" is not present, it implies that it was left at its usual value. I want to make sure we don't log too much for the user.